cedar-policy-core/src/entities.rs

```diff
@@ -823,34 +990,201 @@ mod json_parsing_tests {
         )
     }
 
+    #[test]
+    fn no_expr_escapes1() {
+        let json = serde_json::json!(
+        [
+        {
+            "uid" : r#"test_entity_type::"Alice""#,
+            "attrs": {
+                "bacon": "eggs",
+                "pancakes": [1, 2, 3],
+                "waffles": { "key": "value" },
+                "toast" : { "__extn" : { "fn" : "decimal", "arg" : "33.47" }},
+                "12345": { "__entity": { "type": "test_entity_type", "id": "bob" } },
+                "a b c": { "__extn": { "fn": "ip", "arg": "222.222.222.0/24" } }
+            },
+            "parents": [
+                { "__entity": { "type" : "test_entity_type", "id" : "bob"} },
+                { "__entity": { "type": "test_entity_type", "id": "catherine" } }
+            ]
+        },
+        ]);
+        let eparser: EntityJsonParser<'_> =
+            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
+        let error = eparser.from_json_value(json).err().unwrap().to_string();
+        assert!(
+            error.contains("in uid field of <unknown entity>, expected a literal entity reference"),
+            "{}",
+            error
+        );
+    }
+
+    #[test]
+    fn no_expr_escapes2() {
+        let json = serde_json::json!(
+        [
+        {
+            "uid" : {
+                "__expr" :
+                    r#"test_entity_type::"Alice""#
+            },
+            "attrs": {
+                "bacon": "eggs",
+                "pancakes": [1, 2, 3],
+                "waffles": { "key": "value" },
+                "toast" : { "__extn" : { "fn" : "decimal", "arg" : "33.47" }},
+                "12345": { "__entity": { "type": "test_entity_type", "id": "bob" } },
+                "a b c": { "__extn": { "fn": "ip", "arg": "222.222.222.0/24" } }
+            },
+            "parents": [
+                { "__entity": { "type" : "test_entity_type", "id" : "bob"} },
+                { "__entity": { "type": "test_entity_type", "id": "catherine" } }
+            ]
+        }
+        ]);
+        let eparser: EntityJsonParser<'_> =
+            EntityJsonParser::new(None, Extensions::all_available(), TCComputation::ComputeNow);
+        let error = eparser.from_json_value(json).err().unwrap().to_string();
+        assert!(
+            error.contains("in uid field of <unknown entity>, expected a literal entity reference"),
```

Comment:
would be better to get an error message explicitly talking about the `__expr` escape in this case. Is that possible?